### PR TITLE
[partial] Test coverage for TS!

### PR DESCRIPTION
**Pull request by JaKXz**
First stab reusing some things [from my own starter](https://github.com/JaKXz/webpack-ts-demo/blob/master/karma.conf.js) - not perfect yet because it covers the transpiled code, not the actual `.ts` 

All suggestions are welcome (feel free to add TODOs below), I will be poking at this for a little while:

~~- [ ] cover the actual `.ts` files not the transpiled code.~~
- [x] verify metrics are accurate
- [x] add results to test artifacts

~~- [ ] discuss/enforce a good threshold?~~ can be done in a separate PR
Connected to rangle/rangle-starter#12

### CODE REVIEW

**Comment by JaKXz** [comment]
Also investigating `nyc`: bcoe/nyc#176

File: karma.conf.js [hunk 1]
```diff
@@ -32,31 +34,38 @@ module.exports = function (config) {
...
-          { test: /\.html$/, loader: 'raw' },
-          { test: /\.css$/, loader: 'style-loader!css-loader' },
-          { test: /\.svg/, loader: 'url' },
-          { test: /\.eot/, loader: 'url' },
-          { test: /\.woff/, loader: 'url' },
-          { test: /\.woff2/, loader: 'url' },
-          { test: /\.ttf/, loader: 'url' },
+          loaders.ts,
+          loaders.html,
+          loaders.css
```
Comment (SethDavenport) [inline karma.conf.js]: What about font loaders - are they not needed?

Comment (SethDavenport) [inline karma.conf.js]: Never mind...

**Comment by SethDavenport** [comment]
Looks pretty good - is it working or is there more that needs to be done?

**Comment by JaKXz** [comment]
@SethDavenport still working on the checklist, but thanks :) 

Might be almost there with `nyc` - we'll see soon enough.

**Comment by SethDavenport** [comment]
Oh right fair enough.  As for the threshold - we'll need to beef up the unit tests, this is already tracked as another issue I think.

**Comment by JaKXz** [comment]
can @SethDavenport @igor-ka review?

File: src/utils/is-promise.test.ts [hunk 2]
```diff
@@ -18,9 +18,9 @@ describe('isPromise', () => {
     const badPayload3 = 'hello world';
     const badPayload4 = 'hello world';
 
-    chai.expect(!isPromise({ promise: badPayload1 }));
-    chai.expect(!isPromise({ promise: badPayload2 }));
-    chai.expect(!isPromise({ promise: badPayload3 }));
-    chai.expect(!isPromise({ promise: badPayload4 }));
+    chai.expect(isPromise({ promise: badPayload1 })).to.be.false;
```
Comment (SethDavenport) [inline src/utils/is-promise.test.ts]: :+1:  I wonder if this is lurking in other starters... if so, can you file an issue?

**Comment by SethDavenport** [comment]
:shipit:  Thanks - feel like patching this over to the ng1 starter?

The react starters take a slightly different approach to unit testing - no karma, running mocha through babel-node and ts-node respectively, using jsdom.  This is because we may want to make them isomorphic soon.

Babel-node + istanbul works fine, ts-node + istanbul I couldn't quite get working (surprise, surprise).  Although I did merge in a new version of ts-node so it might be worth trying again.

**Pull request merged**